### PR TITLE
Rest api metrics

### PR DESCRIPTION
**Pull request by RyanLassigBanks**

### CODE REVIEW

File: ci/sawtooth-build-docs [hunk 1]
```diff
@@ -71,7 +71,9 @@ RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sou
  && rm -rf /var/lib/apt/lists/* \
  && pip3 install \
     pylint \
-    bandit
+    bandit \
+&& pip3  install git+git://github.com/omergertel/pyformance.git@de7bee8ceddbb1c3ea3b6c28373a5b066b04b3b7 \
```
Comment (vaporos) [inline ci/sawtooth-build-docs]: This should be an apt-get install of the previously built pyformance deb.

File: docker/sawtooth-dev-python [hunk 2]
```diff
@@ -86,8 +86,9 @@ RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sou
  && pip3 install \
     pylint \
     bandit \
-    pyformance \
-    coverage --upgrade
+    coverage --upgrade \
```
Comment (vaporos) [inline docker/sawtooth-dev-python]: Remove this change, it is already correct after we add the built deb to the repo.

File: docker/sawtooth-int-rest-api [hunk 3]
```diff
@@ -41,7 +41,11 @@ RUN cd /debs \
  && echo "deb file:/debs ./" >> /etc/apt/sources.list \
  && echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sources.list \
  && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 8AA7AF1F1091A5FD \
- && apt-get update
+ && apt-get update \
+ && apt-get install -y -q python3-pip  git\
+ && pip3 install git+git://github.com/omergertel/pyformance.git@de7bee8ceddbb1c3ea3b6c28373a5b066b04b3b7 \
```
Comment (vaporos) [inline docker/sawtooth-int-rest-api]: Use deb package instead.

File: rest_api/sawtooth_rest_api/rest_api.py [hunk 4]
```diff
@@ -118,6 +129,12 @@ def start_rest_api(host, port, connection, timeout):
     app.router.add_get('/subscriptions', subscriber_handler.subscriptions)
     app.on_shutdown.append(lambda app: subscriber_handler.on_shutdown())
 
+    # Start metrics reporter
+
```
Comment (vaporos) [inline rest_api/sawtooth_rest_api/rest_api.py]: Remove the blank line between this command and the line below it.

File: rest_api/sawtooth_rest_api/rest_api.py [hunk 5]
```diff
@@ -177,11 +196,26 @@ def main():
...
+        reporter = None
+        if rest_api_config.opentsdb_url:
+            url = urlparse(rest_api_config.opentsdb_url)
+            proto, db_server, db_port, = url.scheme, url.hostname, url.port
+
+            registry = MetricsRegistry()
+            reporter = InfluxReporter(
+                registry=registry,
+                reporting_interval=1,
+                database=rest_api_config.opentsdb_db,
```
Comment (vaporos) [inline rest_api/sawtooth_rest_api/rest_api.py]: Do we properly handle opentsdb_url being set but opentsdb_db not being set?

Comment (RyanLassigBanks) [reply rest_api/sawtooth_rest_api/rest_api.py]: If opentsdb_url is None it defaults to localhost

File: rest_api/sawtooth_rest_api/rest_api.py [hunk 6]
```diff
@@ -177,11 +195,26 @@ def main():
...
 
+        reporter = None
+        if rest_api_config.opentsdb_url:
+            url = urlparse(rest_api_config.opentsdb_url)
+            proto, db_server, db_port, = url.scheme, url.hostname, url.port
+
+            registry = MetricsRegistry()
+            reporter = InfluxReporter(
+                registry=registry,
+                reporting_interval=1,
```
Comment (jsmitchell) [inline rest_api/sawtooth_rest_api/rest_api.py]: This reporting interval is probably a bit aggressive. We're doing every 10 seconds on the telegraf config.

**Pull request merged**